File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-eq-shared-5-4/ReturnSwapAmount.cpp
// ReturnSwapAmount.cpp 
#include "ReturnSwapAmount.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

ReturnSwapAmount::ReturnSwapAmount(TiXmlNode* xmlNode)
: LegAmount(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //cashSettlementNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* cashSettlementNode = xmlNode->FirstChildElement("cashSettlement");

   if(cashSettlementNode){cashSettlementIsNull_ = false;}
   else{cashSettlementIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- cashSettlementNode , address : " << cashSettlementNode << std::endl;
   #endif
   if(cashSettlementNode)
   {
      if(cashSettlementNode->Attribute("href") || cashSettlementNode->Attribute("id"))
      {
          if(cashSettlementNode->Attribute("id"))
          {
             cashSettlementIDRef_ = cashSettlementNode->Attribute("id");
             cashSettlement_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(cashSettlementNode));
             cashSettlement_->setID(cashSettlementIDRef_);
             IDManager::instance().setID(cashSettlementIDRef_,cashSettlement_);
          }
          else if(cashSettlementNode->Attribute("href")) { cashSettlementIDRef_ = cashSettlementNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { cashSettlement_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(cashSettlementNode));}
   }

   //optionsExchangeDividendsNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* optionsExchangeDividendsNode = xmlNode->FirstChildElement("optionsExchangeDividends");

   if(optionsExchangeDividendsNode){optionsExchangeDividendsIsNull_ = false;}
   else{optionsExchangeDividendsIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- optionsExchangeDividendsNode , address : " << optionsExchangeDividendsNode << std::endl;
   #endif
   if(optionsExchangeDividendsNode)
   {
      if(optionsExchangeDividendsNode->Attribute("href") || optionsExchangeDividendsNode->Attribute("id"))
      {
          if(optionsExchangeDividendsNode->Attribute("id"))
          {
             optionsExchangeDividendsIDRef_ = optionsExchangeDividendsNode->Attribute("id");
             optionsExchangeDividends_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(optionsExchangeDividendsNode));
             optionsExchangeDividends_->setID(optionsExchangeDividendsIDRef_);
             IDManager::instance().setID(optionsExchangeDividendsIDRef_,optionsExchangeDividends_);
          }
          else if(optionsExchangeDividendsNode->Attribute("href")) { optionsExchangeDividendsIDRef_ = optionsExchangeDividendsNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { optionsExchangeDividends_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(optionsExchangeDividendsNode));}
   }

   //additionalDividendsNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* additionalDividendsNode = xmlNode->FirstChildElement("additionalDividends");

   if(additionalDividendsNode){additionalDividendsIsNull_ = false;}
   else{additionalDividendsIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- additionalDividendsNode , address : " << additionalDividendsNode << std::endl;
   #endif
   if(additionalDividendsNode)
   {
      if(additionalDividendsNode->Attribute("href") || additionalDividendsNode->Attribute("id"))
      {
          if(additionalDividendsNode->Attribute("id"))
          {
             additionalDividendsIDRef_ = additionalDividendsNode->Attribute("id");
             additionalDividends_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(additionalDividendsNode));
             additionalDividends_->setID(additionalDividendsIDRef_);
             IDManager::instance().setID(additionalDividendsIDRef_,additionalDividends_);
          }
          else if(additionalDividendsNode->Attribute("href")) { additionalDividendsIDRef_ = additionalDividendsNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { additionalDividends_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(additionalDividendsNode));}
   }

   //allDividendsNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* allDividendsNode = xmlNode->FirstChildElement("allDividends");

   if(allDividendsNode){allDividendsIsNull_ = false;}
   else{allDividendsIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- allDividendsNode , address : " << allDividendsNode << std::endl;
   #endif
   if(allDividendsNode)
   {
      if(allDividendsNode->Attribute("href") || allDividendsNode->Attribute("id"))
      {
          if(allDividendsNode->Attribute("id"))
          {
             allDividendsIDRef_ = allDividendsNode->Attribute("id");
             allDividends_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(allDividendsNode));
             allDividends_->setID(allDividendsIDRef_);
             IDManager::instance().setID(allDividendsIDRef_,allDividends_);
          }
          else if(allDividendsNode->Attribute("href")) { allDividendsIDRef_ = allDividendsNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { allDividends_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(allDividendsNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeBoolean> ReturnSwapAmount::getCashSettlement()
{
   if(!this->cashSettlementIsNull_){
        if(cashSettlementIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(cashSettlementIDRef_));
        }else{
             return this->cashSettlement_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<XsdTypeBoolean> ReturnSwapAmount::getOptionsExchangeDividends()
{
   if(!this->optionsExchangeDividendsIsNull_){
        if(optionsExchangeDividendsIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(optionsExchangeDividendsIDRef_));
        }else{
             return this->optionsExchangeDividends_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<XsdTypeBoolean> ReturnSwapAmount::getAdditionalDividends()
{
   if(!this->additionalDividendsIsNull_){
        if(additionalDividendsIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(additionalDividendsIDRef_));
        }else{
             return this->additionalDividends_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<XsdTypeBoolean> ReturnSwapAmount::getAllDividends()
{
   if(!this->allDividendsIsNull_){
        if(allDividendsIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(allDividendsIDRef_));
        }else{
             return this->allDividends_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-eq-shared-5-4/ReturnSwapAmount.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class ReturnSwapAmount : LegAmount
    {
        public ReturnSwapAmount(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode cashSettlementNode = xmlNode.SelectSingleNode("cashSettlement");
            
            if (cashSettlementNode != null)
            {
                if (cashSettlementNode.Attributes["href"] != null || cashSettlementNode.Attributes["id"] != null) 
                {
                    if (cashSettlementNode.Attributes["id"] != null) 
                    {
                        cashSettlementIDRef_ = cashSettlementNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(cashSettlementNode);
                        IDManager.SetID(cashSettlementIDRef_, ob);
                    }
                    else if (cashSettlementNode.Attributes["href"] != null)
                    {
                        cashSettlementIDRef_ = cashSettlementNode.Attributes["href"].Value;
                    }
                    else
                    {
                        cashSettlement_ = new XsdTypeBoolean(cashSettlementNode);
                    }
                }
                else
                {
                    cashSettlement_ = new XsdTypeBoolean(cashSettlementNode);
                }
            }
            
        
            XmlNode optionsExchangeDividendsNode = xmlNode.SelectSingleNode("optionsExchangeDividends");
            
            if (optionsExchangeDividendsNode != null)
            {
                if (optionsExchangeDividendsNode.Attributes["href"] != null || optionsExchangeDividendsNode.Attributes["id"] != null) 
                {
                    if (optionsExchangeDividendsNode.Attributes["id"] != null) 
                    {
                        optionsExchangeDividendsIDRef_ = optionsExchangeDividendsNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(optionsExchangeDividendsNode);
                        IDManager.SetID(optionsExchangeDividendsIDRef_, ob);
                    }
                    else if (optionsExchangeDividendsNode.Attributes["href"] != null)
                    {
                        optionsExchangeDividendsIDRef_ = optionsExchangeDividendsNode.Attributes["href"].Value;
                    }
                    else
                    {
                        optionsExchangeDividends_ = new XsdTypeBoolean(optionsExchangeDividendsNode);
                    }
                }
                else
                {
                    optionsExchangeDividends_ = new XsdTypeBoolean(optionsExchangeDividendsNode);
                }
            }
            
        
            XmlNode additionalDividendsNode = xmlNode.SelectSingleNode("additionalDividends");
            
            if (additionalDividendsNode != null)
            {
                if (additionalDividendsNode.Attributes["href"] != null || additionalDividendsNode.Attributes["id"] != null) 
                {
                    if (additionalDividendsNode.Attributes["id"] != null) 
                    {
                        additionalDividendsIDRef_ = additionalDividendsNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(additionalDividendsNode);
                        IDManager.SetID(additionalDividendsIDRef_, ob);
                    }
                    else if (additionalDividendsNode.Attributes["href"] != null)
                    {
                        additionalDividendsIDRef_ = additionalDividendsNode.Attributes["href"].Value;
                    }
                    else
                    {
                        additionalDividends_ = new XsdTypeBoolean(additionalDividendsNode);
                    }
                }
                else
                {
                    additionalDividends_ = new XsdTypeBoolean(additionalDividendsNode);
                }
            }
            
        
            XmlNode allDividendsNode = xmlNode.SelectSingleNode("allDividends");
            
            if (allDividendsNode != null)
            {
                if (allDividendsNode.Attributes["href"] != null || allDividendsNode.Attributes["id"] != null) 
                {
                    if (allDividendsNode.Attributes["id"] != null) 
                    {
                        allDividendsIDRef_ = allDividendsNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(allDividendsNode);
                        IDManager.SetID(allDividendsIDRef_, ob);
                    }
                    else if (allDividendsNode.Attributes["href"] != null)
                    {
                        allDividendsIDRef_ = allDividendsNode.Attributes["href"].Value;
                    }
                    else
                    {
                        allDividends_ = new XsdTypeBoolean(allDividendsNode);
                    }
                }
                else
                {
                    allDividends_ = new XsdTypeBoolean(allDividendsNode);
                }
            }
            
        
        }
        
    
        #region CashSettlement_
        private XsdTypeBoolean cashSettlement_;
        public XsdTypeBoolean CashSettlement_
        {
            get
            {
                if (this.cashSettlement_ != null)
                {
                    return this.cashSettlement_; 
                }
                else if (this.cashSettlementIDRef_ != null)
                {
                    cashSettlement_ = IDManager.getID(cashSettlementIDRef_) as XsdTypeBoolean;
                    return this.cashSettlement_; 
                }
                else
                {
                      return this.cashSettlement_; 
                }
            }
            set
            {
                if (this.cashSettlement_ != value)
                {
                    this.cashSettlement_ = value;
                }
            }
        }
        #endregion
        
        public string cashSettlementIDRef_ { get; set; }
        #region OptionsExchangeDividends_
        private XsdTypeBoolean optionsExchangeDividends_;
        public XsdTypeBoolean OptionsExchangeDividends_
        {
            get
            {
                if (this.optionsExchangeDividends_ != null)
                {
                    return this.optionsExchangeDividends_; 
                }
                else if (this.optionsExchangeDividendsIDRef_ != null)
                {
                    optionsExchangeDividends_ = IDManager.getID(optionsExchangeDividendsIDRef_) as XsdTypeBoolean;
                    return this.optionsExchangeDividends_; 
                }
                else
                {
                      return this.optionsExchangeDividends_; 
                }
            }
            set
            {
                if (this.optionsExchangeDividends_ != value)
                {
                    this.optionsExchangeDividends_ = value;
                }
            }
        }
        #endregion
        
        public string optionsExchangeDividendsIDRef_ { get; set; }
        #region AdditionalDividends_
        private XsdTypeBoolean additionalDividends_;
        public XsdTypeBoolean AdditionalDividends_
        {
            get
            {
                if (this.additionalDividends_ != null)
                {
                    return this.additionalDividends_; 
                }
                else if (this.additionalDividendsIDRef_ != null)
                {
                    additionalDividends_ = IDManager.getID(additionalDividendsIDRef_) as XsdTypeBoolean;
                    return this.additionalDividends_; 
                }
                else
                {
                      return this.additionalDividends_; 
                }
            }
            set
            {
                if (this.additionalDividends_ != value)
                {
                    this.additionalDividends_ = value;
                }
            }
        }
        #endregion
        
        public string additionalDividendsIDRef_ { get; set; }
        #region AllDividends_
        private XsdTypeBoolean allDividends_;
        public XsdTypeBoolean AllDividends_
        {
            get
            {
                if (this.allDividends_ != null)
                {
                    return this.allDividends_; 
                }
                else if (this.allDividendsIDRef_ != null)
                {
                    allDividends_ = IDManager.getID(allDividendsIDRef_) as XsdTypeBoolean;
                    return this.allDividends_; 
                }
                else
                {
                      return this.allDividends_; 
                }
            }
            set
            {
                if (this.allDividends_ != value)
                {
                    this.allDividends_ = value;
                }
            }
        }
        #endregion
        
        public string allDividendsIDRef_ { get; set; }
        
    
        
    
    }
    
}

